### PR TITLE
Allocate memory for input tensor

### PR DESCRIPTION
**Pull request by yinghai**
*Description*:
This diff is wrote by @rdzhabarov. It's a mid-step where Glow will take care of the max batch size input. Later this step can be optimized away. 

*Testing*:
unit test. 
*Documentation*:
[Optional Fixes #issue]

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

File: include/glow/Base/Tensor.h [hunk 1]
```diff
@@ -270,6 +270,32 @@ class Tensor final {
     reset(t);
   }
```
Comment (nadavrot) [inline include/glow/Base/Tensor.h]: Doxygen comment please.

File: include/glow/Base/Tensor.h [hunk 2]
```diff
@@ -270,6 +270,32 @@ class Tensor final {
...
 
+  void resize(const Type &T) {
```
Comment (nadavrot) [inline include/glow/Base/Tensor.h]: I don't think that T is a meaningful name.

File: include/glow/Base/Tensor.h [hunk 3]
```diff
@@ -270,6 +270,32 @@ class Tensor final {
...
+    // Do nothing if it's the same shape.
+    if (type_.dims().equals(T.dims())) {
+      return;
```
Comment (nadavrot) [inline include/glow/Base/Tensor.h]: You don't copy the content from the previous tensor if it's the same size?

File: include/glow/Base/Tensor.h [hunk 4]
```diff
@@ -270,6 +270,32 @@ class Tensor final {
...
+    }
+
+    // Allocate memory for new tensor.
```
Comment (nadavrot) [inline include/glow/Base/Tensor.h]: You can use reset()

**Comment by rdzhabarov** [comment]
This was patched to the fbcode to unblock some internal testing scenarios.

I do not think it's ideal to merge it here in the current state, let's make it better there first and then republish.

**Comment by rdzhabarov** [comment]
I'll re-publish once we sort this out internally.

**Pull request closed without merging**